src/xrpld/ledger/detail/View.cpp

```diff
@@ -1875,7 +1875,7 @@ rippleCredit(
             view.update(sle);
         }
         else
-            return tecINTERNAL;
+            return tecMPT_ISSUANCE_NOT_FOUND;
```

Comment:
If a sender is not the issuer, need to check if mpt issuance doesn't exist. I.e., if it doesn't exist then the error is `tecMPT_ISSUANCE_NOT_FOUND` and if mpt token doesn't exist then the error is `tecNO_AUTH`.

Reply:
Could you clarify? The code currently does what you have described.

Reply:
If a sender is not the issuer then we are getting MPToken object. If this object doesn't exist then `tecNO_AUTH` is returned. This is not correct error if `MPTokenIssuance` object doesn't exist. I think we should check at the start of the function if `MPTokenIssuance` doesn't exist and return `tecMPT_ISSUANCE_NOT_FOUND`.

Reply:
i moved the check to the start of function